### PR TITLE
Create pull_request_template.md

### PR DESCRIPTION
**Pull request by holmeso**
In an attempt to ensure that PRs that we submit contain as much useful information as possible so that the reviewer is left in no doubt as to the nature of the change, here is a template that the submitter can follow.
If you have any suggestions/comments/etc. please sing out

### CODE REVIEW

**Comment by holmeso** [comment]
I should mention that this template was copied from [https://embeddedartistry.com/blog/2017/8/4/a-github-pull-request-template-for-your-projects](https://embeddedartistry.com/blog/2017/8/4/a-github-pull-request-template-for-your-projects)

File: pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+- [ ] This change requires a documentation update
+
+# How Has This Been Tested?
+
+Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
+
+- [ ] Test A
+- [ ] Test B
+
+**Test Configuration**:
```
Comment (delocalizer) [inline pull_request_template.md]: Do we need this section?

File: pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+# Checklist:
+
+- [ ] My code follows the style guidelines of this project
+- [ ] I have performed a self-review of my own code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] I have made corresponding changes to the documentation
+- [ ] My changes generate no new warnings
+- [ ] I have added tests that prove my fix is effective or that my feature works
+- [ ] New and existing unit tests pass locally with my changes
+- [ ] Any dependent changes have been merged and published in downstream modules
```
Comment (delocalizer) [inline pull_request_template.md]: I don't know what this means, so I'm not sure if it's relevant for us?

**Comment by holmeso** [comment]
Sorry Conrad, when I made the changes, it rejected your review - would you mind taking another look?

File: pull_request_template.md (hunk 1 above)
Comment (holmeso) [reply pull_request_template.md]: nope - removed

File: pull_request_template.md (hunk 2 above)
Comment (holmeso) [reply pull_request_template.md]: agree - removed

**Pull request merged**